promql/engine.go

```diff
@@ -176,7 +176,21 @@ func (q *query) Exec(ctx context.Context) *Result {
 		span.SetTag(queryTag, q.stmt.String())
 	}
 
+	// Log query in active log
+	queryCompleted := make(chan struct{})
+
+	go func() {
```

Comment:
Why is this in a goroutine?

Reply:
To decouple active query tracking from query execution, and make it faster since it would be run in parallel, if possible. I haven't done any measurements vs. its sequential counterpart though. I'll check which one's faster on my side, and let you know.

Reply:
Keep it simple, the whole query execution will use far more than this.